File: catkin_ws/src/servcpp/src/sumar_cliente.cpp
#include "ros/ros.h"
#include "servcpp/servicios.h"

int main(int argc, char **argv)
{
    ros::init(argc, argv, "sumar_cliente");
    if (argc != 3)
    {
        ROS_INFO("Modo de uso: ~$sumar_cliente a b");
        return 1;
    }
    
    ros::NodeHandle n;
    ros::ServiceClient cliente = n.serviceClient<servcpp::servicios>("suma");

    servcpp::servicios srv;
    srv.request.a = atof(argv[1]);
    srv.request.b = atof(argv[2]);
    if (cliente.call(srv))
    {
        ROS_INFO("Suma: %f +  %f = %f",srv.request.a,srv.request.b, srv.response.c);
    }
    else
    {
        ROS_ERROR("Falla al llamar al servidor Sumador ");
        return 1;
    }

    return 0;
} 

File: catkin_ws/src/servpy/src/sumar_cliente.py
#!/usr/bin/env python
import sys
import rospy
from servpy.srv import *

def sumar_cliente(x, y):
    rospy.wait_for_service('sumar')
    try:
        suma = rospy.ServiceProxy('sumar',servicios)
        respl = suma(x, y)
        return respl.c
    except rospy.ServiceException as e:
        print("Falla al llamar al servidor: %f"%e)

def uso():
    return "Modo de uso %f [a b]"%sys.argv[0]

if __name__ == "__main__":
    if len(sys.argv) == 3:
        x = float(sys.argv[1])
        y = float(sys.argv[2])
    else:
        print(uso())
        sys.exit(1)
    print("Requerimiento %f + %f"%(x, y))
    print("%f + %f = %f"%(x, y, sumar_cliente(x, y)))
